tensorboard/data/server/data_compat.rs

```diff
@@ -405,6 +406,24 @@ mod tests {
         }
     }
 
+    /// Macro to construct `Bytes` from an array of numerics having a `to_le_bytes()` method.
+    ///
+    /// Use like: `let b = to_le_bytes![1.0, 2.0f32];`
+    ///
+    /// Note the suffixed literal, which is necessary so the compiler knows the intended type.
+    //
+    // TODO(Rust 1.51): I think we can avoid intermediate Vecs with array::IntoIter from
+    // https://github.com/rust-lang/rust/pull/80470, e.g. with:
+    //   Bytes::from_iter([].iter().flat_map(|v| array::IntoIter::new(v.to_le_bytes())))
+    macro_rules! to_le_bytes {
+        ($($x:expr),+ $(,)?) => (
+            Bytes::from_iter([$($x),+].iter()
```

Comment:
Good point about pulling in the trait, makes sense.  That said I realized I pulled in `FromIterator` when I actually could just use `From` i.e. I can do `Bytes::from(...same thing...)` which doesn't require pulling in a trait.  (Not sure why I had `FromIterator`, I think in an earlier iteration I still had an iterator at that stage rather than a `Vec`.)

I know doing `.into()` would be more idiomatic than an explicit `from()` but that doesn't work for the `.as_slice()` usage where the type inference breaks down, I get:

    cannot infer type for type parameter `T` declared on the trait `Into`

Hence the explicit `Bytes::from()`.  I could also still do `.into_iter().collect::<Bytes>()` but that seems a little indirect relative to just doing `from()` if you don't have a strong preference.

Reply:
> I know doing `.into()` would be more idiomatic than an explicit `from()`

I think `Bytes::from` is perfectly fine here. The extra clarity of the
target type is helpful since this is a macro context.

Reply:
Okay, done.